pkg/kv/kvserver/mvcc_gc_queue.go

```diff
@@ -118,11 +143,22 @@ func largeAbortSpan(ms enginepb.MVCCStats) bool {
 // single priority. If any task is overdue, shouldQueue returns true.
 type mvccGCQueue struct {
 	*baseQueue
+
+	// Set to true when GC finds range that has a hint indicating that range is
+	// completely cleared.
+	lastRangeWasHighPriority bool
+	// leaseholderCheck is a leasholder check used by high priority replica scanner
+	// its only purpose is to allow test function injection.
+	leaseholderCheck func(ctx context.Context, replica *Replica, timeout time.Duration) error
```

Comment:
If we need this, we should use a testing knob to propagate it. See e.g. how the replicate queue does this:

https://github.com/cockroachdb/cockroach/blob/fc12919071a37f659d9bfdb0c8ba16bb9b46cd41/pkg/kv/kvserver/replicate_queue.go#L547